src/main/scala/shell/xilinx/VC707NewShell.scala

```diff
@@ -136,31 +167,35 @@ class JTAGDebugVC707Overlay(val shell: VC707Shell, val name: String, params: JTA
     } }
   } }
 }
+class JTAGDebugVC707ShellPlacer(val shell: VC707Shell, val shellInput: JTAGDebugShellInput)(implicit val valName: ValName)
+  extends JTAGDebugShellPlacer[VC707Shell] {
+  def place(designInput: JTAGDebugDesignInput) = new JTAGDebugVC707PlacedOverlay(shell, valName.name, designInput, shellInput)
```

Comment:
@terpstra @erikdanie is there a way we can avoid repeating this `def place() = new PlacedOverlay` in every top-level `ShellPlacer` class that's written?

Reply:
I believe the reason that we can't avoid it is that it needs to know what `PlacedOverlay` it is actually creating (i.e. a  `new JTAGDebugVC707Overlay`. If you were to pass it that class as a type parameter and have a default `place` function, you would still need to type the majority of that line anyway to give it the types and the inputs. In that way, I feel like it's a little clearer to just write the whole place function. This also gives you the ability to define a different place function if, though I'm not sure you ever would want to do that instead of just putting different code in the constructor of the overlay that you are `new`-ing.